src/ripper/ripper.cpp

```diff
@@ -38,23 +38,23 @@ const char kWavFileTypeFormatChunk[] = "WAVEfmt ";
 const char kWavDataString[] = "data";
 }  // namespace
 
-Ripper::Ripper(QObject* parent)
+Ripper::Ripper(CdIo_t* cdio, QObject* parent)
     : QObject(parent),
+      cdio_(cdio),
       transcoder_(new Transcoder(this)),
       cancel_requested_(false),
       finished_success_(0),
       finished_failed_(0),
       files_tagged_(0) {
-  cdio_ = cdio_open(NULL, DRIVER_UNKNOWN);
-
+  Q_ASSERT(cdio_);  // TODO: error handling
```

Comment:
This one also need proper handling as of now; I'll wait for a decision on the exception matter before tackling this, though.

Reply:
If cdio_ is null here, then it's a Clementine bug, isn't it? In that case, I think the assert is fine.